File: app/src/main/java/com/jingxi/smartlife/pad/mvp/util/ScreenManager.java
package com.jingxi.smartlife.pad.mvp.util;

import android.net.Uri;
import android.provider.Settings;

import com.wujia.lib_common.utils.AppContext;

/**
 * Created by asus on 2016/12/8.
 * 屏幕亮度调节器
 */
public class ScreenManager {

    /**
     * 获得当前屏幕亮度的模式
     *
     * @return 1 为自动调节屏幕亮度,0 为手动调节屏幕亮度,-1 获取失败
     */
    public static int getScreenMode() {
        int mode = -1;
        try {
            mode = Settings.System.getInt(AppContext.get().getContentResolver(),
                    Settings.System.SCREEN_BRIGHTNESS_MODE);
        } catch (Settings.SettingNotFoundException e) {
            e.printStackTrace();
        }
        return mode;
    }

    /**
     * 获得当前屏幕亮度值
     *
     * @return 0--255
     */
    public static int getScreenBrightness() {
        int screenBrightness = -1;
        try {
            screenBrightness = Settings.System.getInt(AppContext.get().getContentResolver(),
                    Settings.System.SCREEN_BRIGHTNESS);
        } catch (Settings.SettingNotFoundException e) {
            e.printStackTrace();
        }
        return screenBrightness;
    }

    /**
     * 设置当前屏幕亮度的模式
     *
     * @param mode 1 为自动调节屏幕亮度,0 为手动调节屏幕亮度
     */
    public static void setScreenMode(int mode) {
        try {
            Settings.System.putInt(AppContext.get().getContentResolver(),
                    Settings.System.SCREEN_BRIGHTNESS_MODE, mode);
            Uri uri = Settings.System
                    .getUriFor("screen_brightness_mode");
            AppContext.get().getContentResolver().notifyChange(uri, null);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    /**
     * 保存当前的屏幕亮度值，并使之生效
     *
     * @param paramInt 0-255
     */
    public static void setScreenBrightness(int paramInt) {
        Settings.System.putInt(AppContext.get().getContentResolver(),
                Settings.System.SCREEN_BRIGHTNESS, paramInt);
        Uri uri = Settings.System
                .getUriFor("screen_brightness");
        AppContext.get().getContentResolver().notifyChange(uri, null);
    }
}

File: app/src/main/java/com/jingxi/smartlife/pad/mvp/util/ScreenManager.kt
package com.jingxi.smartlife.pad.mvp.util

import android.provider.Settings

import com.wujia.lib_common.utils.AppContext

/**
 * Created by asus on 2016/12/8.
 * 屏幕亮度调节器
 */
object ScreenManager {
    const val LIGHT_MODE_AUTO = 1
    const val LIGHT_MODE_MANUAL = 0
    const val LIGHT_MODE_FAILED = -1


    /**
     * 获得当前屏幕亮度的模式
     *
     * @return 1 为自动调节屏幕亮度,0 为手动调节屏幕亮度,-1 获取失败
     */
    /**
     * 设置当前屏幕亮度的模式
     *
     * @param mode 1 为自动调节屏幕亮度,0 为手动调节屏幕亮度
     */

    var screenMode: Int
        @LightMode
        get() {
            var mode = LIGHT_MODE_FAILED
            try {
                mode = Settings.System.getInt(AppContext.get().contentResolver,
                        Settings.System.SCREEN_BRIGHTNESS_MODE)
            } catch (e: Settings.SettingNotFoundException) {
                e.printStackTrace()
            }

            return mode
        }
        set(@LightMode mode) = try {
            Settings.System.putInt(AppContext.get().contentResolver,
                    Settings.System.SCREEN_BRIGHTNESS_MODE, mode)
            val uri = Settings.System
                    .getUriFor("screen_brightness_mode")
            AppContext.get().contentResolver.notifyChange(uri, null)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    /**
     * 获得当前屏幕亮度值
     *
     * @return 0--255
     */
    /**
     * 保存当前的屏幕亮度值，并使之生效
     *
     * @param paramInt 0-255
     */
    var screenBrightness: Int
        get() {
            var screenBrightness = -1
            try {
                screenBrightness = Settings.System.getInt(AppContext.get().contentResolver,
                        Settings.System.SCREEN_BRIGHTNESS)
            } catch (e: Settings.SettingNotFoundException) {
                e.printStackTrace()
            }

            return screenBrightness
        }
        set(paramInt) {
            Settings.System.putInt(AppContext.get().contentResolver,
                    Settings.System.SCREEN_BRIGHTNESS, paramInt)
            val uri = Settings.System
                    .getUriFor("screen_brightness")
            AppContext.get().contentResolver.notifyChange(uri, null)
        }

}

